### PR TITLE
Added AuthorizationFilter class

### PR DESCRIPTION
**Pull request by aleksandr-hrankin**

### CODE REVIEW

**Review by dariiabutorina — APPROVED** [review]

File: src/main/java/com/internet/shop/controller/LoginController.java [hunk 1]
```diff
@@ -32,6 +33,7 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
             User user = authenticationService.login(login, password);
             HttpSession session = req.getSession();
             session.setAttribute(USER_ID, user.getId());
+            session.setAttribute(USER_ROLES, user.getRoles());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controller/LoginController.java]: why do you need to put user roles in session?

File: src/main/webapp/WEB-INF/view/header.jsp [hunk 2]
```diff
@@ -26,35 +27,48 @@
...
-                            <a class="dropdown-item" href="${pageContext.request.contextPath}/users/all">Users</a>
-                            <a class="dropdown-item" href="${pageContext.request.contextPath}/orders/all">Orders</a>
-                            <div class="dropdown-divider"></div>
-                            <a class="dropdown-item" href="${pageContext.request.contextPath}/inject-data">Inject</a>
-                        </div>
-                    </li>
+<%--                    <c:if test="${not empty sessionScope.user_id}">--%>
+<%--                        <c:forEach var="role" items="${sessionScope.user_role}">--%>
+<%--                            <c:choose>--%>
+<%--                                <c:when test="${role == 'ADMIN'}">--%>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/view/header.jsp]: remove comments

File: src/main/webapp/WEB-INF/view/header.jsp [hunk 3]
```diff
@@ -26,35 +27,48 @@
...
+                                        </li>
+                                        <li class="nav-item">
+                                            <a class="nav-link pr-4" href="${pageContext.request.contextPath}/user/orders">
+                                                <i class="fas fa-box"> Orders</i>
+                                            </a>
+                                        </li>
+<%--                                </c:when>--%>
+<%--                            </c:choose>--%>
+<%--                        </c:forEach>--%>
+<%--                    </c:if>--%>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/view/header.jsp]: same

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        if (isAuthorized(user, protectedUrls.get(url))) {
+            filterChain.doFilter(req, resp);
+        } else {
+            req.getRequestDispatcher("/WEB-INF/view/access-denied.jsp").forward(req, resp);
+        }
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java:68]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**